core/src/main/scala/org/apache/spark/TaskContext.scala

```diff
@@ -94,6 +94,11 @@ abstract class TaskContext extends Serializable {
    */
   def isCompleted(): Boolean
 
+  /**
+   * Returns true if the task has failed.
+   */
+  def isFailed(): Boolean
```

Comment:
Would be great if we can add this into PySpark side as well ...

Reply:
It seems python taskcontext does not have `isCompleted` and `isInterrupted` method. Is there any reason ?

Reply:
We should have them all :-).

Reply:
I don't mind doing separately in a differnet PR

Reply:
I will create a new pr for it later

Reply:
@HyukjinKwon does pyspark support update status from jvm side ? It seems we can only send a snapshot status to python side ?